src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -49,24 +51,34 @@ public SQLServerDataTable() throws SQLServerException {
     /**
      * Clears this data table.
      */
-    public synchronized void clear() {
-        rowCount = 0;
-        columnCount = 0;
-        columnMetadata.clear();
-        columnNames.clear();
-        rows.clear();
+    public void clear() {
+        lock.lock();
+        try {
+            rowCount = 0;
+            columnCount = 0;
+            columnMetadata.clear();
+            columnNames.clear();
+            rows.clear();
+        } finally {
+            lock.unlock();
+        }
     }
 
     /**
      * Returns an iterator on the rows of the data table.
      * 
      * @return an iterator on the rows of the data table.
      */
-    public synchronized Iterator<Entry<Integer, Object[]>> getIterator() {
-        if ((null != rows) && (null != rows.entrySet())) {
-            return rows.entrySet().iterator();
+    public Iterator<Entry<Integer, Object[]>> getIterator() {
+        lock.lock();
+        try {
+            if (null != rows) {
```

Comment:
Why is the check for `rows.entrySet() not equal to null` removed here?

Reply:
The change probably "came out in the wash".  It's a redundant check as rows.entrySet() can never be null.